src/middlewared/middlewared/plugins/system.py

```diff
@@ -277,15 +272,38 @@ def version(self):
         return "BOOTING"
 
     async def __get_license(self):
-        licenseobj = get_license()[0]
-        if not licenseobj:
+        if not os.path.exists(LICENSE_FILE):
+            return
+
+        with open(LICENSE_FILE, 'r') as f:
```

Comment:
William do we want to run this in a coroutine ?

Reply:
Good point, its probably safer to turn this into a method